cmd/bnscli/cmd_key.go

```diff
@@ -35,9 +40,17 @@ created. This command fails if the private key file already exists.
 		return fmt.Errorf("private key file %q already exists, delete this file and try again", *keyPathFl)
 	}
 
-	_, priv, err := ed25519.GenerateKey(nil)
+	mnemonic, err := readInput(input)
 	if err != nil {
-		return fmt.Errorf("cannot generate ed25519 key: %s", err)
+		return fmt.Errorf("cannot read mnemonic: %s", err)
+	}
+
+	// We do not allow for passphrase.
+	seed := bip39.NewSeed(string(mnemonic), "")
```

Comment:
I think here we are missing an English mnemonic validity check. The _From mnemonic to seed_ algorithm works for any string input, so we need to check beforehand.

Reply:
Done in https://github.com/iov-one/weave/pull/923/commits/ef5fb9f53c074f9bd2c6687008e6054dc5fff3ed